tests/test_variant_coordinates.py

```diff
@@ -30,9 +32,33 @@ def test_get_coordinates_range_location(fasta: Fasta):
     ) == ('21', 33341700, 'AGAC', {'DEL': 'A', 'GAC': 'AGAC'})
 
 
-def test_get_coordinates_not_match_reference(fasta: Fasta):
+def test_get_coordinates_not_match_reference_snp(fasta: Fasta):
+    # Multi-allelic SNP, reference not annotated
     assert fasta.get_chr_pos_ref(
         'rs1051266',
-        'NC_000021.9:33341701_33341703',
-        [['TTT', 'TTT'], ['TTT', 'DEL'], ['DEL', 'DEL']]
-    ) == ('21', 33341700, None, None)
+        'NC_000021.9:45537880',
+        [['C', 'C'], ['C', 'G'], ['G', 'G']]
+    ) == ('21', 45537880, 'T', {'C': 'C', 'G': 'G'})
+
+
+def test_get_coordinates_not_match_reference_del(fasta: Fasta):
+    # Deletion
+    # Mock NCBI response, so we can simulate an exact scenario on chr21 without searching for the right RS
+    with patch('opentargets_pharmgkb.variant_coordinates.get_spdi_coords_for_rsid') as m_spdi_coords:
+        m_spdi_coords.return_value = ('NC_000021.9', 45537879, 'TGCTGC', ['TGC'])
+        result = fasta.get_chr_pos_ref(
+            'rs1051266',
+            'NC_000021.9:45537880_45537885',
+            [['TGC', 'TGC'], ['TGC', 'DEL'], ['DEL', 'DEL']]
+        )
+        assert result == ('21', 45537879, 'GTGC', {'TGC': 'GTGC', 'DEL': 'G'})
+
+
+def test_normalise(fasta: Fasta):
+    # Section of chr21 consisting of ATTT repeats
+    assert fasta.get_ref_from_fasta('NC_000021.9', 7678481, 7678480 + (10*4)) \
+           == 'TTTTATTTATTTATTTATTTATTTATTTATTTATTTATTT'
+    # Variant that deletes one repeat from the middle is normalised to be left-aligned
+    assert fasta.normalise('NC_000021.9', 7678489, ['ATTTATTT', 'ATTT']) == ('NC_000021.9', 7678481, ['TTTTA', 'T'])
+    # Same variant with different representation (empty allele)
+    assert fasta.normalise('NC_000021.9', 7678489, ['ATTT', '']) == ('NC_000021.9', 7678481, ['TTTTA', 'T'])
```

Comment:
Have you tried with more than two alleles ? One REF +2 ALTs

Reply:
I think I did at some point, but I've now added this test